src/credentials/activation.ts

```diff
@@ -52,10 +52,9 @@ export async function loginWithMostRecentCredentials(
     } else if (
         providerMap &&
         profileNames.length === 1 &&
-        (await manager.getCredentialsProvider(providerMap[profileNames[0]]))!.canAutoConnect()
+        (await manager.getCredentialsProvider(providerMap[profileNames[0]]))!.canAutoConnect() &&
+        (await loginManager.login({ passive: true, providerId: providerMap[profileNames[0]] })) // Auto-connect if there is exactly one profile.
```

Comment:
Seems weird to move an action into the if statement criteria. Have a specific reason for this?

Reply:
Looked nicer to me, but now I think think about it there may be times where you would want to act based off a failed login. I'll revert.